geopandas/_vectorized.py

```diff
@@ -474,23 +474,26 @@ def is_simple(data):
 
 
 def is_ring(data):
+    warnings.warn(
+        "is_ring currently returns True for Polygons, which is not correct. "
+        "This will be corrected to False in a future release.",
```

Comment:
We could also only raise this warning if there are actually polygons in the data? 
That adds some extra slowdown (but checking types is quite cheap), but would be a nicer user experience I think? Eg you can already use it for linestrings without getting the warning